dataset-processor-python/data_loader.py

```diff
@@ -1,55 +1,272 @@
+        "cell_thumbnail",
+        "cell_image",
+        "parent_thumbnail",
+        "parent_image",
+    ]
+
+    def __init__(self, data_loader, inputs):
+        # initialize the json files in the data folder
+        self.data = data_loader.data
+        self.inputs = inputs
+        self.cell_feature_analysis_data = []
+        self.feature_defs_data = []
+        self.dataset_data = {}
+        # utility variables - to organize and categorize the feature data for file writing
+        self.feature_def_keys = set()
+        self.discrete_features_dict = {}
+        self.features_data_order = []
+
+    @staticmethod
+    def convert_str_to_num(value):
+        try:
+            numeric_value = float(value)
+            if numeric_value.is_integer():
+                return int(numeric_value)
+            return numeric_value
+        except ValueError:
+            # if not number, return the original value
+            return value
+
+    @staticmethod
+    def is_valid_feature_value(value):
+        return pd.isna(value) or isinstance(value, (int, float))
 
-    def get_inputs(self):
+    @staticmethod
+    def is_discrete(numeric_value):
         """
-        get the inputs from the user
+        Determines if the numeric value is discrete
         """
-        # version is required, title and description are optional, default values are set as empty string
-        # return version, title, description
-        pass
+        return not pd.isna(numeric_value) and numeric_value == int(numeric_value)
 
-    def create_dataset_folder(self):
+    def update_feature_metadata(self, column, value):
+        is_discrete = self.is_discrete(value)
+        if (
+            column in self.discrete_features_dict
+            and is_discrete != self.discrete_features_dict[column]
+        ):
+            print(
+                f"Column {column} has both discrete and continuous values. Please make sure that all values in a column are either discrete or continuous."
+            )
+            return
+        self.discrete_features_dict[column] = is_discrete
+        self.feature_def_keys.add(column)
+
+    def process_features(self, column, value, features):
+        if self.is_valid_feature_value(value):
+            features.append(value)
+            self.update_feature_metadata(column, value)
+        else:
+            print(f"Invalid value: {value} in column {column}")
+            return
+
+    @staticmethod
+    def get_unit(key):
         """
-        create a folder with the database_name
-        initialize the folder with required json files
+        Extracts the unit from the key if present
         """
-        # folder_name = self.dataset_name + "_" + version
-        # path = os.path.join("data", folder_name)
-        # os.makedirs(path, exist_ok=True)
-        # create json files -> feature_defs.json, dataset.json, cell_feature_analysis.json, image_settings.json
-        pass
+        match = re.search(r"\((.*?)\)", key)
```

Comment:
```suggestion
        match = re.search(r"\((.*?)\)$", key)
```
This change *should* fix the bug I described earlier, where it'll only match with the very last closing parentheses. Might need some additional testing but this should work!